File: backend/app/schemas/hank_preferences.py
"""Typed personal presentation choices; never authority or production policy."""

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.base import UTCModel


class HankPreferenceValues(BaseModel):
    model_config = ConfigDict(extra='forbid')
    briefing_detail: Literal['concise', 'standard'] = 'standard'
    focus_area: Literal['role_default', 'my_work', 'shop', 'quality', 'purchasing', 'inventory', 'shipping'] = (
        'role_default'
    )
    handoff_format: Literal['bullets', 'checklist'] = 'bullets'
    follow_up_alerts: bool = True


class HankPreferenceCommand(BaseModel):
    model_config = ConfigDict(extra='forbid')
    expected_company_id: int = Field(gt=0)
    expected_version: int = Field(ge=0)


class HankPreferenceSave(HankPreferenceCommand):
    preferences: HankPreferenceValues


class HankPreferenceResponse(UTCModel):
    company_id: int
    version: int
    preferences: HankPreferenceValues
    updated_at: datetime | None = None
    can_edit: bool

File: frontend/src/types/hankPreferences.ts
export interface HankPreferencesValues {
  briefing_detail: 'concise' | 'standard';
  focus_area: 'role_default' | 'my_work' | 'shop' | 'quality' | 'purchasing' | 'inventory' | 'shipping';
  handoff_format: 'bullets' | 'checklist';
  follow_up_alerts: boolean;
}

export interface HankPreferencesResponse {
  company_id: number;
  version: number;
  preferences: HankPreferencesValues;
  updated_at: string | null;
  can_edit: boolean;
}

export interface HankPreferencesCommand {
  expected_company_id: number;
  expected_version: number;
}

export interface HankPreferencesUpdate extends HankPreferencesCommand {
  preferences: HankPreferencesValues;
}

export const DEFAULT_HANK_PREFERENCES: HankPreferencesValues = {
  briefing_detail: 'standard',
  focus_area: 'role_default',
  handoff_format: 'bullets',
  follow_up_alerts: true,
};
